### PR TITLE
Adding import functionality from old news sites.

### PR DESCRIPTION
**Pull request by paustian**
This code is pretty simple, but it gets the job done.

### CODE REVIEW

File: MU/NewsModule/Controller/MessageController.php [hunk 1]
```diff
@@ -248,4 +248,75 @@ public function handleSelectedEntriesAction(
...
+            $message->setSlug($item['urltitle']);
+            $message->setNotes($item['notes']);
+            $message->setAmountOfViews($item['counter']);
+            $message->setMessageLanguage($item['language']);
+            $message->setWorkflowState('approved');
+            $entityManager->persist($message);
+        }
+        //save back to the database.
+        $entityManager->flush();
+        $this->addFlash('status', $this->__("$num_Items Messages Imported."));
```
Comment (Guite) [inline MU/NewsModule/Controller/MessageController.php:319]: This is not a valid usage of Gettext. It should be something like `$this->addFlash('status', $this->__f('%amount% messages Imported.', ['%amount%' => $num_Items]));`

File: MU/NewsModule/Controller/MessageController.php [hunk 2]
```diff
@@ -248,4 +248,75 @@ public function handleSelectedEntriesAction(
...
+     * @param Request $request Current request instance
+     *
+     *
+     * @throws RuntimeException Thrown if executing the workflow action fails
+     */
+
+    public function importNewsArticlesAction(Request $request){
+        //check permission, only admins can do this.
+        if (!$this->hasPermission( '::', '::', ACCESS_ADMIN)) {
+            throw new AccessDeniedException($this->__('You do not have pemission to import news items. Admin access is needed.'));
```
Comment (Guite) [inline MU/NewsModule/Controller/MessageController.php:265]: typo: `pemission`

File: MU/NewsModule/Controller/MessageController.php [hunk 3]
```diff
@@ -248,4 +248,75 @@ public function handleSelectedEntriesAction(
...
+     *
+     * @param Request $request Current request instance
+     *
+     *
+     * @throws RuntimeException Thrown if executing the workflow action fails
+     */
+
+    public function importNewsArticlesAction(Request $request){
+        //check permission, only admins can do this.
+        if (!$this->hasPermission( '::', '::', ACCESS_ADMIN)) {
```
Comment (Guite) [inline MU/NewsModule/Controller/MessageController.php:264]: How about `if (!$this->hasPermission('MUNewsModule::', '::', ACCESS_ADMIN)) {`?

**Comment by Guite** [comment]
Nice!
Added comments about some minor things I've found.

Can you also please revert/remove these changes https://github.com/zikula-modules/News/pull/48/files as part of your PR because they are replaced by it?

File: MU/NewsModule/Controller/MessageController.php [hunk 4]
```diff
@@ -248,4 +248,75 @@ public function handleSelectedEntriesAction(
...
+        //I am taking a shortcut here since the approver of past messages may not be on the site anymore
+        //and instead I am using the current ID of the person that is importing the messages who is the admin.
+        $currentUserApi = $this->get('zikula_users_module.current_user');
+        $author_id = $currentUserApi->get('uid');
+        $author =  $currentUserApi->get('uname');
+        $user = $entityManager->getRepository('Zikula\UsersModule\Entity\UserEntity')->findByUids($author_id);
+        //walk the array and fill in the fields in the new data that are needed.
+        foreach($items as $item){
+            $message = new MessageEntity();
+            $cr_date = new \DateTime($item['cr_date']);
```
Comment (Guite) [inline MU/NewsModule/Controller/MessageController.php:291]: Please handle the last updated date similarly: `$lu_date = new \DateTime($item['lu_date']);` ... `$message->setUpdatedDate($lu_date);`

File: MU/NewsModule/Controller/MessageController.php (hunk 3 above)
Comment (paustian) [reply MU/NewsModule/Controller/MessageController.php:264]: Works for me

**Comment by paustian** [comment]
I will take out those changes to install and resubmit the pull request. Is that the way to do it?

**Comment by Guite** [comment]
Just push any further changes into your master branch. The pull request will update automatically then.

**Comment by paustian** [comment]
Ok. I went into smart git paustian/News and did a push. Is it all set now? Sorry, it took a while for me to respond, teaching summer classes has kept me busy.

**Comment by Guite** [comment]
No additional commits have been added to your repository.

**Comment by Guite** [comment]
Thanks @paustian :+1: 

I've added some minor cleanup patches in addition to your changes.

**Pull request merged**